src/azure-cli-core/azure/cli/core/azclierror.py

```diff
@@ -4,85 +4,178 @@
 # --------------------------------------------------------------------------------------------
 
 import sys
-from enum import Enum
 
+import azure.cli.core.telemetry as telemetry
 from knack.util import CLIError
 from knack.log import get_logger
 
 logger = get_logger(__name__)
+# pylint: disable=unnecessary-pass
 
 
-class AzCLIErrorType(Enum):
-    """ AzureCLI error types """
-
-    # userfaults
-    CommandNotFoundError = 'CommandNotFoundError'
-    ArgumentParseError = 'ArgumentParseError'
-    ValidationError = 'ValidationError'
-    ManualInterrupt = 'ManualInterrupt'
-    # service side error
-    ServiceError = 'ServiceError'
-    # client side error
-    ClientError = 'ClientError'
-    # unexpected error
-    UnexpectedError = 'UnexpectedError'
-
-
+# region: Base Layer
+# Base class for all the AzureCLI defined error classes.
+# DO NOT raise the error class here directly in your codes.
 class AzCLIError(CLIError):
-    """ AzureCLI error definition """
-
-    def __init__(self, error_type, error_msg, raw_exception=None, command=None):
-        """
-        :param error_type: The name of the AzureCLI error type.
-        :type error_type: azure.cli.core.util.AzCLIErrorType
-        :param error_msg: The error message detail.
-        :type error_msg: str
-        :param raw_exception: The raw exception.
-        :type raw_exception: Exception
-        :param command: The command which brings the error.
-        :type command: str
-        :param recommendations: The recommendations to resolve the error.
-        :type recommendations: list
-        """
-        self.error_type = error_type
+    """ Base class for all the AzureCLI defined error classes. """
+
+    def __init__(self, error_msg):
```

Comment:
Better to support `recommendation` in the constructor so that the exception can be created and raised in one line:

```py
raise AzureCLIError(error_msg, recommendation)
```

Reply:
sure